### ISSUE TITLE
Examples don't work.

### ISSUE THREAD

**Issue by eusahn** [rticommunity/rticonnextdds-connector-js#157]
Running => `node ./reader_web.js `

````
rticonnextdds-connector-js/node_modules/ffi-napi/lib/dynamic_library.js:67
    if (match = err.match(/^(([^ \t()])+\.so([^ \t:()])*):([ \t])*/)) {
                    ^
TypeError: Cannot read properties of null (reading 'match')
````

**Comment by Alxe** [comment]
Hello @eusahn, this has been identified as a consequence of mixing two dependencies, `ref-napi` and `ref-struct-napi`. [As noted in the documentation of `ref-napi`](https://github.com/node-ffi-napi/ref-napi?tab=readme-ov-file#incompatible-packages), this is an incompatible mix.

We've tracked this internally as CON-299 and identified that it will happen for Node 18.8 and upwards. Resolution is scheduled to be done for a next release of Connector but, if you're using Connector from source and not from NPM, you can consider manually applying the following patch.

```
diff --git a/examples/nodejs/simple/writer.js b/examples/nodejs/simple/writer.js
index 7751652..1022afd 100644
--- a/examples/nodejs/simple/writer.js
+++ b/examples/nodejs/simple/writer.js
@@ -6,7 +6,7 @@
 * This code contains trade secrets of Real-Time Innovations, Inc.             *
 ******************************************************************************/
 
-const sleep = require('sleep')
+const msleep = require('util').promisify(setTimeout)
 const path = require('path')
 const rti = require('rticonnextdds-connector')
 const configFile = path.join(__dirname, '/../ShapeExample.xml')
@@ -26,7 +26,7 @@ const run = async () => {
       output.instance.setString('color', 'BLUE')
       output.write()
 
-      sleep.msleep(500)
+      msleep(500)
     }
 
     console.log('Exiting...')
diff --git a/package.json b/package.json
index aef49b7..6d30c08 100644
--- a/package.json
+++ b/package.json
@@ -16,8 +16,7 @@
     "events": "^3.2.0",
     "ffi-napi": "^3.1.0",
     "ref-napi": "^3.0.1",
-    "ref-struct-napi": "^1.1.1",
-    "sleep": "^6.3.0"
+    "ref-struct-di": "^1.1.1"
   },
   "scripts": {
     "test": "mocha ./test/nodejs/",
diff --git a/rticonnextdds-connector.js b/rticonnextdds-connector.js
index 3ade42b..ef42dbb 100644
--- a/rticonnextdds-connector.js
+++ b/rticonnextdds-connector.js
@@ -10,7 +10,7 @@ const os = require('os')
 const ref = require('ref-napi')
 const ffi = require('ffi-napi')
 const path = require('path')
-const StructType = require('ref-struct-napi')
+const StructType = require('ref-struct-di')(ref)
 const EventEmitter = require('events').EventEmitter
 
 /**
``

**Comment by Alxe** [comment]
This should've been addressed by dd0dd91.
